File: erupt-core/src/main/java/com/erupt/model/EruptApiModel.java
package com.erupt.model;

/**
 * Created by liyuepeng on 10/9/18.
 */
public class EruptApiModel {

    private Status status;

    private Object data;

    public enum Status {
        //200
        SUCCESS(200),
        //500
        ERROR(500),
        //403
        NO_LOGIN(403),
        //405
        NO_RIGHT(405);

        public int code;

        Status(int code) {
            this.code = code;
        }
    }
}

File: erupt-web-build/src/app/erupt/model/erupt-api-model.ts
/**
 * Created by liyuepeng on 11/1/18.
 */
export interface EruptApiModel {
  data: any;
  status: Status;
}

export enum Status {
  SUCCESS = "SUCCESS",
  ERROR = "ERROR",
  NO_LOGIN = "NO_LOGIN",
  NO_RIGHT = "NO_RIGHT"
}

